### ISSUE TITLE
Add proper deadlock detector to prototype rust SDK

### ISSUE THREAD

**Issue by bergundy** [temporalio/sdk-core#166]
Yeah so this can't stay as-is then, clearly. It's still a nice thing to be able to detect, though. Maybe we can upgrade this to also check that there is nothing that needs to be unblocked. *then* it's a deadlock. If that's easy go for it, if not we can fix in another PR

_Originally posted by @Sushisource in https://github.com/temporalio/sdk-core/pull/163#r685388622_

**Comment by p44ytwz25w-cloud** [comment]
campbell-erin
